src/solver/conv_MP_anydirectional_winograd.cpp

```diff
@@ -56,66 +58,167 @@
+    const auto& out_H = (is_wrw) ? (params).kernel_size_h : (params).out_height;                   \
+    const auto& out_W = (is_wrw) ? (params).kernel_size_w : (params).out_width;
+
+#define DEFINE_SHADER_ALIASES(params) \
+    DEFINE_SHADER_ALIASES_2((params), (params).direction.IsBackwardWrW())
 
 #if MIOPEN_BACKEND_HIP
-#define GENERATE_MAIN_OPTIONS(options)                                     \
-    GenerateClangDefsym((options), "acc_type", 1);                         \
-    GenerateClangDefsym((options), "ROCM_METADATA_VERSION", 5);            \
-    GenerateClangDefsym((options), "xformx_o_size", WinoDataW);            \
-    GenerateClangDefsym((options), "xformy_o_size", WinoDataH);            \
-    GenerateClangDefsym((options), "xformx_d_size", wino_xform_w);         \
-    GenerateClangDefsym((options), "xformy_d_size", wino_xform_h);         \
-    GenerateClangDefsym((options), "xformx_f_size", WinoFilterW);          \
-    GenerateClangDefsym((options), "xformy_f_size", WinoFilterH);          \
-    GenerateClangDefsym((options), "fdilation_w", params.kernel_stride_w); \
-    GenerateClangDefsym((options), "fdilation_h", params.kernel_stride_h);
+
+#define DEFINE_GETDTILEHWSIZE(params)                                                         \
+    const auto                                                                                \
+        wino_dtile_h =                                                                        \
+            solver::ConvMPAnydirectWinograd<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW>:: \
+                GetSolverWinoDtileHWSize(params, 1),                                          \
+        wino_dtile_w =                                                                        \
+            solver::ConvMPAnydirectWinograd<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW>:: \
+                GetSolverWinoDtileHWSize(params, 0);
+
+#define DEFINE_SHADER_CONV_MOD_ALIASES_4(params, is_fwd, is_bwd, is_wrw) \
+    const int stride_h     = (is_fwd) ? (params).kernel_stride_h : 1;    \
+    const int stride_w     = (is_fwd) ? (params).kernel_stride_w : 1;    \
+    const int f_dilation_h = (is_wrw) ? (params).kernel_stride_h : 1;    \
+    const int f_dilation_w = (is_wrw) ? (params).kernel_stride_w : 1;    \
+    const int d_dilation_h = (is_bwd) ? (params).kernel_stride_h : 1;    \
+    const int d_dilation_w = (is_bwd) ? (params).kernel_stride_w : 1;
+
+#define DEFINE_SHADER_CONV_MOD_ALIASES(params)                            \
+    DEFINE_SHADER_CONV_MOD_ALIASES_4((params),                            \
+                                     (params).direction.IsForward(),      \
+                                     (params).direction.IsBackwardData(), \
+                                     (params).direction.IsBackwardWrW())
+
+#define GENERATE_MAIN_OPTIONS(options)                   \
+    KernelBuildParameters options{};                     \
```

Comment:
It would be better if you would move the declaration outside the macro at least. Otherwise, the macro is still unsafe in usage. Placing it in if would for sure not work as one may expect now. And then you may change it to function because it doesn't declare anything.

Reply:
Although if you don't want to copypaste the declaration, I guess, it's kind of ok, I won't stop you from merging this.